common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -254,8 +256,7 @@ public  <T> T createPredictPayload(Map<String, String> parameters) {
         if (predictAction.isPresent() && predictAction.get().getRequestBody() != null) {
             String payload = predictAction.get().getRequestBody();
             payload = fillNullParameters(parameters, payload);
-            StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
-            payload = substitutor.replace(payload);
+            payload = applyCustomSubstitution(parameters, payload);
             if (!isJson(payload)) {
```

Comment:
In my local testing, I found this can return false on "valid" json.  The issue was that parameter values needed to be escaped.  I can make a separate PR for that, but thought Id mention it here since there's related code in this PR for escaping parameter values.

Reply:
In my testing I saw, at this point payload is supposed to replaced with the actual value:

like: `"{ \"texts\": ${parameters.texts}, \"truncate\": ${parameters.truncate} }"`  --> `"{ \"texts\": [\"hello world\"], \"truncate\": \"END\" }"` which is a valid JSON.

Could you please share your use case where you get false for a "valid" JSON?

Reply:
I am basically taking a bunch of text from my OpenSearch cluster (as search results/hits) and passing it as part of the `messages` parameter to OpenAI (search hit -> message).  I don't know which character in my text caused the issue, but to get around it, I applied a patch locally to always do `escapeJson` on the messages and got around the issue.  So, what I am trying to say is that if the text is not properly escaped for Json, you can get IllegalArgumentException.  I guess this is not a comment specific to your PR.  I am not pointing out an issue with the custom substitution logic per se.  Let me know if you want me to open a new issue to track this.

Reply:
@austintlee I think you are using 2.x branch. We have some bug fix in 2.9 branch, we just backported them to 2.x branch. Please rebase your code and retest.